realm/src/main/java/io/realm/RealmObject.java

```diff
@@ -78,9 +85,14 @@
     protected Row row;
     protected Realm realm;
 
+    private final List<RealmChangeListener> listeners = new CopyOnWriteArrayList<RealmChangeListener>();
```

Comment:
Adding all this state here probably means we decide to keep `RealmObject` around. Otherwise it will have to be moved to the annotation processor. See https://github.com/realm/realm-java/issues/1036

At this point there is probably so many upsides to having this compared to the negative, but we should take the discussion.